rmw_opensplice_cpp/src/rmw_subscription.cpp

```diff
@@ -226,9 +226,7 @@ rmw_create_subscription(
   if (subscriber_info) {
     rmw_free(subscriber_info);
   }
-  if (subscription) {
-    rmw_subscription_free(subscription);
-  }
```

Comment:
Wouldn't this be the case when the goto from the following line triggers: https://github.com/ros2/rmw_opensplice/pull/207/files#diff-153e779bdc552cb4e36e9029830b2307R197